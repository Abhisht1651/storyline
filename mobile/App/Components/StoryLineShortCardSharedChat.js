// @flow
import React from 'react'
import { View, Text, Image, ImageProps } from 'react-native'

// Styles
import styles from './Styles/StoryLineShortCardSharedChatStyle'
import images from '../Themes/Images'

type Props = {
  sender: string,
  heading: string,
  reach: number,
  storyTitle: string,
  trending: boolean,
  storyImage: ImageProps
}

const storyLineShortCardSharedChat = (props: Props) => {
  const { sender, heading, reach, storyTitle, trending, storyImage } = props
  const { iconStoryLineTrending, iconStoryLineReach } = images

  return (
    <View style={styles.container}>
      <View style={styles.senderContainer}>
        <Text style={styles.senderText}>{sender} send you a storyline!</Text>
      </View>
      <View style={styles.divider} />
      <View style={styles.heading}>
        <Text style={styles.headingText}>{heading}</Text>
        {trending ? <View style={styles.dot} /> : null}
        {trending ? (
          <Image source={iconStoryLineTrending} style={styles.iconStoryLineTrending} />
        ) : null}
        {trending ? <Text style={styles.headingText}> Trending</Text> : null}
      </View>
      <View style={styles.reachContainer}>
        <Image source={iconStoryLineReach} style={styles.iconStoryLineReach} />
        <Text style={styles.reachContainerText}> Reached {reach}K people</Text>
      </View>
      <Text style={styles.newsTitleText}>{storyTitle}</Text>
      <View style={styles.imageContainer}>
        <Image source={storyImage} style={styles.image} />
      </View>
    </View>
  )
}

storyLineShortCardSharedChat.defaultProps = {
  sender: 'Sophie',
  heading: 'World',
  reach: 45,
  storyTitle: 'Brexit: UK exit from the European Union',
  trending: true,
  storyImage: images.storyImage
}

export default storyLineShortCardSharedChat
