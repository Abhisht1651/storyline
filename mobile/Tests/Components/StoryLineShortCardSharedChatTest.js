// @flow
import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import StoryLineShortCardSharedChat from '../../App/Components/StoryLineShortCardSharedChat'

describe('StoryLineShortCardSharedChat', () => {
  it('should render the component itself correctly', () => {
    const wrapper = shallow(<StoryLineShortCardSharedChat />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<StoryLineShortCardSharedChat />)
    expect(wrapper.find('Text')).toHaveLength(5)
  })
  it('should render the Image Component', () => {
    const wrapper = shallow(<StoryLineShortCardSharedChat />)
    expect(wrapper.find('Image')).toHaveLength(3)
  })
})
