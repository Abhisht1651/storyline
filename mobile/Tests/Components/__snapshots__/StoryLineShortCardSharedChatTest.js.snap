// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryLineShortCardSharedChat should render the component itself correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E6ECED80",
      "borderRadius": 30,
      "minHeight": 301,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "opacity": 1,
        "width": "88%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Regular",
          "fontSize": 32,
          "letterSpacing": -0.11,
          "marginVertical": 34,
          "opacity": 1,
          "textAlign": "left",
        }
      }
    >
      Sophie
       send you a storyline!
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#DCDCE9",
        "borderBottomWidth": 1,
        "opacity": 1,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 30,
        "width": "88%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Regular",
          "fontSize": 28,
          "letterSpacing": -0.08,
          "lineHeight": 32,
          "opacity": 1,
          "textAlign": "left",
        }
      }
    >
      World
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#1F1F60",
          "borderRadius": 2,
          "height": 4,
          "marginHorizontal": 10,
          "width": 4,
        }
      }
    />
    <Image
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "height": 34,
          "width": 34,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Regular",
          "fontSize": 28,
          "letterSpacing": -0.08,
          "lineHeight": 32,
          "opacity": 1,
          "textAlign": "left",
        }
      }
    >
       Trending
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 10,
        "width": "88%",
      }
    }
  >
    <Image
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "height": 26,
          "width": 26,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8F8FAF",
          "fontFamily": "BananaGrotesk-Regular",
          "fontSize": 24,
          "letterSpacing": -0.07,
          "lineHeight": 28,
          "opacity": 1,
          "textAlign": "left",
        }
      }
    >
       Reached 
      45
      K people
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#1F1F60",
        "fontFamily": "BananaGrotesk-Regular",
        "fontSize": 32,
        "letterSpacing": -0.11,
        "lineHeight": 44,
        "marginTop": 14,
        "opacity": 1,
        "textAlign": "left",
        "width": "88%",
      }
    }
  >
    Brexit: UK exit from the European Union
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center",
        "height": 256,
        "marginBottom": 30,
        "marginTop": 8,
        "width": "88%",
      }
    }
  >
    <Image
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "borderRadius": 15,
          "height": "100%",
          "opacity": 1,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
